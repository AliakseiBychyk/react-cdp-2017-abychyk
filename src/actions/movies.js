export const RECEIVE_MOVIES = 'RECEIVE_MOVIES'

export const receiveMovies = (movies) => ({
  type: RECEIVE_MOVIES,
  movies  
})

export const fetchMoviesJson = (criterion, query) => { 
  const url = `https://api.themoviedb.org/3/search/${criterion}?api_key=${api_key}&query=${query}&include_adult=true`   

  return fetch(url, {
    method: 'GET'
  }).then(resp => resp.json())
    .then(data => {
      return criterion === 'movie'
        ? data.results
        : data.results[0].known_for
    })
    .catch(console.error)
}