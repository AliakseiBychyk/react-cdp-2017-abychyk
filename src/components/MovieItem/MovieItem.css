.movieItem {
  text-align: left;
  display: block;
  height: 400px;
  padding: 30px;
  margin: auto;
  background-color: slategray;
}

.backToSearchButton {
  display: inline-block;
  float: right;
}

.details {

}

.img {
  display: inline-block;
  vertical-align: top;
  padding: 10px;
  height: 300px;
}

.text {
  display: inline-block;
  vertical-align: top;
  padding: 10px;
  width: 50%;
}

.summary {

}

.director {

}

.cast {

}