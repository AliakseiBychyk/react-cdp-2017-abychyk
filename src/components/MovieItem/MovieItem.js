import React from 'react'
import { Link } from 'react-router-dom'
import { Button } from 'react-bootstrap'
import styles from './MovieItem.css'

const MovieItem = ({movie}) => (
  movie 
  ? <div className={styles.movieItem}>
      <div className={styles.backToSearchButton}>
        <Link to='/'>  
          <Button>SEARCH</Button> 
        </Link>  
      </div>
      <div className={styles.details}>
        <img
          src={movie.poster}
          alt={movie.show_title}
          className={styles.img}
        />
        <div className={styles.text}>
          <div>
            <p3 className={styles.title}>{movie.show_title}</p3>
            <div>{movie.rating}</div>
          </div>
          <div>
            <div>{movie.release_year}</div>
            <div>{movie.runtime}</div>
          </div>
          <div className={styles.summary}>
            {movie.summary}  
          </div>
          <div className={styles.director}>
            Director: {movie.director}  
          </div>
          <div className={styles.cast}>
            Cast: {movie.show_cast}  
          </div>
        </div>
      </div>
    </div>
  : null  
)

export default MovieItem