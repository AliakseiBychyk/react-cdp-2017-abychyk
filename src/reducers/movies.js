import { RECEIVE_MOVIES } from '../actions/movies'

export default (state = [], action) => {
  switch (action.type) {
    
    case RECEIVE_MOVIES:
      return action.movies

    default:
      return state
  }
}