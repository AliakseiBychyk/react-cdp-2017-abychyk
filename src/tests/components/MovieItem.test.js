import React from 'react'
import { shallow } from 'enzyme'
import MovieItem from '../../components/MovieItem/MovieItem'
import movies from '../fixtures/testingMovieQuery'

test('should render MovieItem with movie', () => {
  const wrapper = shallow(<MovieItem movie={movies[2]}/>)
  expect(wrapper).toMatchSnapshot()
})