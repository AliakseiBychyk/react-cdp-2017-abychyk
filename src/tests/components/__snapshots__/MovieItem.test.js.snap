// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieItem with movie 1`] = `
<div
  className="movieItem"
>
  <div
    className="backToSearchButton"
  >
    <Link
      replace={false}
      to="/"
    >
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        disabled={false}
      >
        SEARCH
      </Button>
    </Link>
  </div>
  <div
    className="details"
  >
    <img
      alt="Terminator 2: Judgment Day"
      className="img"
      src="http://netflixroulette.net/api/posters/60028202.jpg"
    />
    <div
      className="text"
    >
      <div>
        <p3
          className="title"
        >
          Terminator 2: Judgment Day
        </p3>
        <div>
          3.9
        </div>
      </div>
      <div>
        <div>
          1991
        </div>
        <div>
          137 min
        </div>
      </div>
      <div
        className="summary"
      >
        A reprogrammed Terminator arrives from the future to protect young John Connor from a relentless shape-shifting cyborg in this sequel to Terminator. On the run, John and his guardian seek John's mother, Sarah, who is locked away in an insane asylum.
      </div>
      <div
        className="director"
      >
        Director: 
        James Cameron
      </div>
      <div
        className="cast"
      >
        Cast: 
        Arnold Schwarzenegger, Linda Hamilton, Robert Patrick, Edward Furlong, Earl Boen, Joe Morton, S. Epatha Merkerson, Castulo Guerra, Danny Cooksey, Jenette Goldstein
      </div>
    </div>
  </div>
</div>
`;
